node/node_v2.go

```diff
@@ -0,0 +1,126 @@
+// These v2 methods are implemented in this separate file to keep the code organized.
+// Note that there is no NodeV2 type and these methods are implemented in the existing Node type.
+
+package node
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+
+	"github.com/Layr-Labs/eigenda/api/clients"
+	"github.com/Layr-Labs/eigenda/core"
+	corev2 "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/gammazero/workerpool"
+)
+
+type requestMetadata struct {
+	blobShardIndex int
+	quorum         core.QuorumID
+}
+type relayRequest struct {
+	chunkRequests []*clients.ChunkRequestByRange
+	metadata      []*requestMetadata
+}
+type response struct {
+	metadata []*requestMetadata
+	bundles  [][]byte
+}
+
+type RawBundles struct {
+	BlobCertificate *corev2.BlobCertificate
+	Bundles         map[core.QuorumID][]byte
+}
+
+func (n *Node) DownloadBatch(ctx context.Context, batch *corev2.Batch) ([]*corev2.BlobShard, []*RawBundles, error) {
```

Comment:
It looks odd to supply the batch as input parameter, while the goal is to produce (download) a batch (unless the parameter is also used to bring out the output which isn't the case here)

Reply:
Agree with the confusion. I'll rename this method to `DownloadBundles`. Is that more clear?